tools/travis-build

```diff
@@ -94,6 +94,28 @@ rm -rf /tmp/pip
 pip install $P4C_PIP_PACKAGES
 # ! ------  END Python2 ----------------------------------------------------
 
+# ! ------  BEGIN VALIDATION -----------------------------------------------
+# These steps are necessary to validate the correct compilation of the P4C test
+# suite programs. See also https://github.com/p4gauntlet/p4_tv.
+# Grab the toz3 extension for the p4 compiler
+mkdir -p /p4c/extensions
+git clone -b stable https://github.com/p4gauntlet/toz3 /p4c/extensions/toz3
+
+# Also pull Gauntlet for translation validation
+git clone -b stable https://github.com/p4gauntlet/p4_tv /gauntlet
+rm -rf /gauntlet/p4c
+ln -s /p4c /gauntlet/p4c
+
+# Install Gauntlet Python dependencies locally
+# Unfortunately we need Python 3.6, which Xenial does not have
```

Comment:
what's so special about 3.6?

Reply:
[f-strings](https://www.python.org/dev/peps/pep-0498/). I have been using them a lot in the code because they are nicer to read and use. Python 3.6 is available with 16.10 but apparently not 16.04.